test/language/expressions/assignment/destructuring/array-elem-init-order.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    Initializer values should be assigned in left-to-right order.
+es6id: 12.14.5.3
+---*/
+
+var value = [];
+var x = 0;
+var result;
+
+result = [ a = x += 1, b = x *= 2 ] = value;
```

Comment:
Please add `var a, b;` for strict-mode compatibility (or mark test as noStrict).